feature-libs/asm/components/asm-bind-cart/asm-bind-cart.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { AsmFacade } from '@spartacus/asm/root';
+import { ActiveCartFacade, MultiCartFacade } from '@spartacus/cart/base/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  OCC_CART_ID_CURRENT,
+  User,
+} from '@spartacus/core';
+import { UserAccountFacade } from '@spartacus/user/account/root';
+import { defer, Subscription } from 'rxjs';
+import { finalize } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-asm-bind-cart',
+  templateUrl: './asm-bind-cart.component.html',
+})
+export class AsmBindCartComponent implements OnInit, OnDestroy {
+  @Input() customer: User;
+  cartId: FormControl = new FormControl('', [
+    Validators.required,
+    Validators.minLength(1),
+  ]);
+
+  loading = false;
+
+  protected subscription = new Subscription();
+
+  constructor(
+    protected globalMessageService: GlobalMessageService,
+    protected asmFacade: AsmFacade,
+    protected activeCartFacade: ActiveCartFacade,
+    protected multiCartFacade: MultiCartFacade,
+    protected userAccountFacade: UserAccountFacade
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.userAccountFacade.get().subscribe((user) => {
```

Comment:
`this.userAccountFacade.get()` can be moved to command. We can make `customerId` in `BindCartParams` as optional. So, only pass `cartId` to `bindCart` function. In command, something like this:
```
protected bindCartCommand$: Command<BindCartParams> =
    this.commandService.create((options: BindCartParams) =>
      options.customerId
        ? this.asmConnector.bindCart(options)
        : this.userAccountFacade.get().pipe(
            filter(isNotUndefined),
            map((user) => user.uid ?? ''),
            switchMap((customerId) =>
              this.asmConnector.bindCart({
                customerId,
                cartId: options.cartId,
              })
            )
          )
    );
      this.asmConnector.bindCart(options)
    );
```

Reply:
The use of `UserAccountFacade` has been encapsulated in the use of a command.  I thought the suggestion was difficult to follow, so I used a different implementation.  Please review and provide any feedback.

Reply:
After review and discussion with @WeizhengSap, here is the feedback.
The parameters passed to the Commands is not really the issue here after all.

The intent behind the Command pattern is for it to be used by facade implementations.

In general, the component should be calling a facade method that may be implemented with a command, but not create commands directly.  In this case, it's ok for the facade method to accept just the cart id and fetch the current user id from within its implementation.   Many facade methods operate on the currently signed in user and retrieve it themselves.

Consider the usage of `UserIdService.takeUserId` to retrieve the userId.  This is what many (perhaps most) facade implementations use.  There are many examples in the code base.